src/Lean/Elab/Binders.lean

```diff
@@ -531,13 +531,13 @@ def elabLetDeclAux (id : Syntax) (binders : Array Syntax) (typeStx : Syntax) (va
     if useLetExpr then
       withLetDecl id.getId type val fun x => do
         addLocalVarInfo id x
-        let body ← elabTerm body expectedType?
+        let body ← elabTermEnsuringType body expectedType?
```

Comment:
@leodemoura Could you double-check that this still accepts all relevant `let`s? I believe that if we posit that in `let_delayed x := v; b` `b` must type-check *without* access to `v`, the change should be correct. Otherwise we might need to fall back to `elabTerm` in that case.